stl/inc/ranges

```diff
@@ -652,7 +652,7 @@ namespace ranges {
         /* [[no_unique_address]] */ _Ty _Value{};
     };
 
-    template <_Destructible_object _Ty>
+    template <_Destructible_object _Ty, bool _Needs_operator_bool = false>

```

Comment:
IIRC we need to flip the default for this to work correctly with #1980 - @miscco is that right? (I have a mild preference for `operator bool` to be opt out, and a slightly stronger preference to avoid interactions between these two PRs if we can.)

Reply:
I could easily be convinced that the default should be opt in, but then I'd like to see this merged into #2042 instead of main,